### PR TITLE
OOP school library: refactor the code

### PR DESCRIPTION
**Pull request by Salwa99**
# OOP school library: refactor the code

### Hey there! :wave: 

**In this PR we :**

1. Respect YAGNI, KISS, and DRY rules and Ruby style guide.
2. Follow SOLID principles: refactor App class and Main class.
3. Follow GitFlow.
4. Fix Linter errors.

_Cheers :sparkler:_

### CODE REVIEW

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 1]
```diff
@@ -26,7 +26,11 @@ def list_people
       puts 'There are no people.'
       return
     end
-    @people.each do |p|
+    display_people(@people)
+  end
+
```
[the remark this answers is not in the document]
Comment (Salwa99) [reply app.rb:31]: Got it @gemmen29 , I'll fix it. Thanks!

[the remark this answers is not in the document]
Comment (rloterh) [reply app.rb:32]: resolved! ✔️

**Comment by AmanWorku** [comment]
@rloterh Sure we will do that. Thank you sir.

**Pull request merged**